tests/custom_selector_name.rs

```diff
@@ -0,0 +1,59 @@
+// This test is the same as basic.rs but with custom context selectors
+
+use snafu::{ResultExt, Snafu};
+use std::{
+    fs, io,
+    path::{Path, PathBuf},
+};
+
+#[derive(Debug, Snafu)]
+enum Error {
+    #[snafu(
+        context(OpenConfigContext),
+        display = r#"("Could not open config file at {}: {}", filename.display(), source)"#
+    )]
+    OpenConfig {
+        filename: PathBuf,
+        source: io::Error,
+    },
+    #[snafu(
+        context(SaveConfigContext),
+        display = r#"("Could not open config file at {}", source)"#
+    )]
+    SaveConfig { source: io::Error },
+    #[snafu(
+        context(InvalidUserContext),
+        display = r#"("User ID {} is invalid", user_id)"#
+    )]
```

Comment:
This style is fine, which is why it compiles, but I just find it ugly for whatever reason. FWIW, I tend to split it into two:

```rust
    #[snafu(context(InvalidUserContext))]
    #[snafu(display = r#"("User ID {} is invalid", user_id)"#)]
```